mm2src/coins/eth.rs

```diff
@@ -2558,6 +2515,90 @@ impl EthCoin {
             }
         }
     }
+
+    fn search_for_swap_tx_spend(
+        &self,
+        tx: &[u8],
+        swap_contract_address: Address,
+        search_from_block: u64,
+    ) -> Result<Option<FoundSwapTxSpend>, String> {
+        let unverified: UnverifiedTransaction = try_s!(rlp::decode(tx));
+        let tx = try_s!(SignedEthTx::new(unverified));
+
+        let func_name = match self.coin_type {
+            EthCoinType::Eth => "ethPayment",
+            EthCoinType::Erc20 { .. } => "erc20Payment",
+        };
+
+        let payment_func = try_s!(SWAP_CONTRACT.function(func_name));
+        let decoded = try_s!(payment_func.decode_input(&tx.data));
+        let id = match &decoded[0] {
+            Token::FixedBytes(bytes) => bytes.clone(),
+            _ => panic!(),
+        };
+
+        let mut current_block = try_s!(self.current_block().wait());
+        if current_block < search_from_block {
+            current_block = search_from_block;
+        }
+
+        let mut from_block = search_from_block;
+
+        loop {
+            let to_block = current_block.min(from_block + 1000);
```

Comment:
Oh, those the magic numbers :D
I think it's ok to use `1000` blocks until there is no ETH fork that mines block much faster than mainnet

Reply:
It's a good point, I will put the `logs_block_range` into the conf file with the default value of 1000 :slightly_smiling_face:

Reply:
The BSC nodes have a parameter `--rangelimit` that accepts no values, it just limits the range of eth_getlogs to 5000 blocks. BSC has a blocktime of 3s, and i guess and hope there will be nothing faster any time soon :-)